doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+; field                               = value
+STATE                                 = "enabled" / "disabled"      ; user enable/disable
+FULL-NAME                             = STRING                      ; Full-name/Description of user
+HASHED-PASSWORD                       = STRING                      ; Hashed password
+HASHED-PASSWORD_HISTORY               = STRING                      ; List of old passwords (0-99)
+PASSWD_DATE_OF_LAST_CHANGE            = UINT32                      ; date of password last change
+ROLE                                  = "admin" / "monitor"         ; Role/Capability
+
+```
+
+```
+; Defines schema for role configuration attributes in ROLE_TABLE:
+key                                   = "<role name>"               ;Role configuration
+; field                               = value
+PRIMARY_GROUP                         = STRING                      ; Primary linux group
+SECONDARY_GROUP                       = STRING                      ; List of secondary linux groups
+```
+
+###  3.3 <a name='CLIYANGmodel'></a>CLI/YANG model
+
+##### YANG model
+```yang
+
+module sonic-user-mgmt {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-user-mgmt";
+	prefix user-mgmt;
+
+    description "User Management YANG Module for SONiC OS";
+
+	revision 2022-12-01 {
+        description "First Revision";
+    }
+   container sonic-user-mgmt {
+
+        container USER_TABLE {
+
+            description "USER MANAGEMENT part of config_db.json";
+
+            list USER_TABLE_LIST {
+
+                key "name";
+
+                leaf name {
+                  type string {
+                        pattern '[a-zA-Z_]([a-zA-Z0-9_-]{0,31}|[a-zA-Z0-9_-]{0,30}$)';
+                        length 1..32 {
+                              error-message "Invalid length for the username.";
+                              error-app-tag username-invalid-length;
+                        }				  
+				          }
+                  description "User name.";
+                }                
+
+                leaf state {
+                    default "enabled";
+                    type string {
+                        pattern "enabled|disabled";
+                    }
+                }
+                
+                leaf role {
+                    default "admin";
+                    type string {
+                        pattern "admin|monitor";
+                    }
+                }                
+                
+                leaf full-name {
+                    type string;
+                }
+                
+                leaf hashed-password {
+                    mandatory true;
+                    type string;
+                }
+                
+                leaf password_history {
+                    type string;
+                  	description "List of hashed passwords seperated by comma."; 
+                }
+
+                leaf passwd_date_of_last_change {
```

Comment:
I don't think so
This field is actually the Unix epoch time of the last password change
It will be used to preserve ageing after upgrade, and it won't be exposed to user.
so it is better to keep it in the original format.